coursework/backend/src/main/java/com/krusty/crab/controller/AnalyticsController.java

```diff
@@ -17,33 +24,85 @@
 @RestController
 @RequiredArgsConstructor
 @Slf4j
+@PreAuthorize("hasRole('Manager')")
 public class AnalyticsController implements AnalyticsApi {
-    
+
     private final AnalyticsService analyticsService;
-    
+    private final ReportViewMapper reportViewMapper;
+
     @Override
     public ResponseEntity<SalesSummary> getSalesSummary(
             @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) OffsetDateTime from,
             @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) OffsetDateTime to) {
+        UserPrincipal user = SecurityUtil.getCurrentUser();
```

Comment:
Если где-то есть util в имени -- это вызывает кринж

```suggestion
        UserPrincipal user = SecurityContext.getCurrentUser();
```